core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/provision/handler/CreateExternalDatabaseHandler.java

```diff
@@ -0,0 +1,130 @@
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DetailedStackStatus;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.database.DatabaseAvailabilityType;
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.core.flow2.externaldatabase.StackUpdaterService;
+import com.sequenceiq.cloudbreak.core.flow2.externaldatabase.ExternalDatabaseService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.reactor.api.event.externaldatabase.CreateExternalDatabaseFailed;
+import com.sequenceiq.cloudbreak.reactor.api.event.externaldatabase.CreateExternalDatabaseRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.externaldatabase.CreateExternalDatabaseResult;
+import com.sequenceiq.cloudbreak.service.environment.EnvironmentClientService;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class CreateExternalDatabaseHandler implements EventHandler<CreateExternalDatabaseRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateExternalDatabaseHandler.class);
+
+    @Inject
+    private EventBus eventBus;
+
+    @Inject
+    private ExternalDatabaseService provisionService;
+
+    @Inject
+    private StackUpdaterService stackUpdaterService;
+
+    @Inject
+    private EnvironmentClientService environmentClientService;
+
+    @Override
+    public String selector() {
+        return "CreateExternalDatabaseRequest";
+    }
+
+    @Override
+    public void accept(Event<CreateExternalDatabaseRequest> createExternalDatabaseRequest) {
+        LOGGER.debug("In CreateExternalDatabaseHandler.accept");
+        CreateExternalDatabaseRequest request = createExternalDatabaseRequest.getData();
+        Stack stack = request.getStack();
+        DatabaseAvailabilityType externalDatabase = ObjectUtils.defaultIfNull(stack.getExternalDatabase(), DatabaseAvailabilityType.NONE);
+        Selectable result;
+        try {
+            if (externalDatabase == DatabaseAvailabilityType.NONE) {
+                LOGGER.info("External database for stack {} is not requested.", stack.getName());
+                result = new CreateExternalDatabaseResult(stack.getId(), EXTERNAL_DATABASE_WAIT_SUCCESS_EVENT.event(), stack.getName(), null);
+            } else {
+                stackUpdaterService.updateStatus(stack.getId(), DetailedStackStatus.EXTERNAL_DATABASE_CREATION_IN_PROGRESS,
+                        ResourceEvent.CLUSTER_EXTERNAL_DATABASE_CREATION_STARTED, "External database creation in progress");
```

Comment:
some log would be nice here

Reply:
logging added here and to terminate handler as well